lib/honey/boilerplates.ex

```diff
@@ -346,82 +361,93 @@ defmodule Honey.Boilerplates do
+          (*array_pool)[#{arg}.value.syscalls_enter_kill_args.pos_syscall_nr] = (Generic){.type = INTEGER, .value.integer = ctx_arg->syscall_nr};
         }
-        if (#{Enum.at(config.func_args, 0)}.value.syscalls_enter_kill_args.pos_pid < ARRAY_POOL_SIZE)
+        if (#{arg}.value.syscalls_enter_kill_args.pos_pid < ARRAY_POOL_SIZE)
         {
-          (*array_pool)[#{Enum.at(config.func_args, 0)}.value.syscalls_enter_kill_args.pos_pid] = (Generic){.type = INTEGER, .value.integer = ctx_arg->pid};
+          (*array_pool)[#{arg}.value.syscalls_enter_kill_args.pos_pid] = (Generic){.type = INTEGER, .value.integer = ctx_arg->pid};
         }
-        if (#{Enum.at(config.func_args, 0)}.value.syscalls_enter_kill_args.pos_sig < ARRAY_POOL_SIZE)
+        if (#{arg}.value.syscalls_enter_kill_args.pos_sig < ARRAY_POOL_SIZE)
         {
-          (*array_pool)[#{Enum.at(config.func_args, 0)}.value.syscalls_enter_kill_args.pos_sig] = (Generic){.type = INTEGER, .value.integer = ctx_arg->sig};
-        }"
+          (*array_pool)[#{arg}.value.syscalls_enter_kill_args.pos_sig] = (Generic){.type = INTEGER, .value.integer = ctx_arg->sig};
+        }
+        """
     end
   end
 
-  def get_ending_main_code(return_var_name) do
-    "if (#{return_var_name}.type != INTEGER)
-    {
+  def generate_ending_main_code(return_var_name) do
+    gen """
+    if (#{return_var_name}.type != INTEGER) {
       op_result = (OpResult){.exception = 1, .exception_msg = \"(IncorrectReturn) eBPF function is not returning an integer.\"};
       goto CATCH;
     }
     return #{return_var_name}.value.integer;
 
-  CATCH:
-    bpf_printk(\"** %s\\n\", op_result.exception_msg);
-    return 0;
-    "
+    CATCH:
+      bpf_printk(\"** %s\\n\", op_result.exception_msg);
+      return 0;
+    """
   end
 
-  def get_license(config) do
-    "char LICENSE[] SEC(\"license\") = \"#{config.license}\";\n\n"
+  def generate_license(config) do
+    gen """
+    char LICENSE[] SEC("license") = "#{config.license}";
+
+
+    """
   end
 
-  def get_main_arguments(config) do
+  def generate_main_arguments(config) do
     case config.libbpf_prog_type do
       "tracepoint/syscalls/sys_enter_kill" ->
         "struct syscalls_enter_kill_args *ctx_arg"
 
       _ ->
-      ""
+        ""
     end
   end
 
-  def get_main(config) do
-    "SEC(\"#{config.libbpf_prog_type}\")\n" <>
-    "int main_func(#{get_main_arguments(config)}) {" <>
-    get_beggning_main_code() <>
-    get_middle_main_code(config) <>
-    "\n// =============== beggining of user code ===============\n" <>
-    config.translated_code.code <>
-    "\n// =============== end of user code ===============\n" <>
-    get_ending_main_code(config.translated_code.return_var_name) <>
-    "}"
+  def generate_main(config) do
+    gen """
+    SEC("#{config.libbpf_prog_type}")
+    int main_func(#{generate_main_arguments(config)}) {
+      #{beggning_main_code()}
```

Comment:
Type in word "beginning"

Reply:
Thanks for pointing this out salfum! We have committed a fix.